packages/i18n-calypso/src/use-translate.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+export default function( i18n ) {
+	const translate = i18n.translate.bind( i18n );
+	const getLocaleSlug = i18n.getLocaleSlug.bind( i18n );
+
+	return function useTranslate() {
+		const [ localeSlug, setLocaleSlug ] = React.useState( getLocaleSlug );
+
+		React.useEffect(() => {
+			const onChange = () => setLocaleSlug( getLocaleSlug() );
+			i18n.on( 'change', onChange );
+			return () => i18n.off( 'change', onChange );
+		}, []);
+
+		return [ translate, localeSlug ];
```

Comment:
I'm not sure I'd use an array here; the two properties aren't as closely-related as the `[ property, setProperty ]` pair returned by `useState`, and they likely won't need to be renamed either. I'd probably return an object instead.

Still, this is really a matter of personal taste here, so happy either way.

Reply:
Another option is to just return `translate`:
```js
const translate = useTranslate();
```
Components never really use anything else. The `localeSlug` needs to be part of state, because the `translate` function is always the same and _something_ needs to change to trigger a rerender. That will be especially true if we switch to React Context instead of one `i18n` listener per each mounted localized component.

I returned array mainly because it's an existing convention for several hooks. But thinking about it a little more, it doesn't look like a good fit for this particular hook.

Reply:
+1 to just returning `translate`

Reply:
Would it be possible to make the language slug available as `translate.localeSlug`?